templates/tower-project.yaml

```diff
@@ -0,0 +1,338 @@
+    AllowedValues:
+      - Enabled
+      - Disabled
+    Default: Disabled
+
+  LifecycleDataTransition:
+    Type: Number
+    Description: >-
+      Number of days until S3 objects are moved to LifecycleDataStorageClass
+    Default: 30
+
+  LifecycleDataStorageClass:
+    Type: String
+    Description: S3 bucket objects will transition into this storage class
+    AllowedValues:
+      - DEEP_ARCHIVE
+      - INTELLIGENT_TIERING
+      - STANDARD_IA
+      - ONEZONE_IA
+      - GLACIER
+    Default: GLACIER
+
+  LifecycleDataExpiration:
+    Type: Number
+    Description: >-
+      Number of days (from creation) when objects are
+      deleted from S3 and the LifecycleDataStorageClass
+    Default: 365000
+
+  AdminArns:
+    Type: List<String>
+    Description: >-
+      List of IAM user and role ARNs (strings) who should have access to 
+      project-specific resources (e.g., private S3 bucket, KMS key, secrets). 
+  
+  NonAdminArns:
+    Type: List<String>
+    Description: >-
+      List of IAM user and role ARNs (strings) who shouldn't have access to 
+      project-specific resources (e.g., private S3 bucket, KMS key, secrets). 
+
+  TemplateRootUrl:
+    Type: String
+    Description: URL of S3 bucket where templates are deployed
+    ConstraintDescription: Must be a valid S3 HTTP URL
+
+  TowerForgePolicyArn:
+    Type: String
+    Description: ARN for Tower Forge IAM policy
+
+  TowerLaunchPolicyArn:
+    Type: String
+    Description: ARN for Tower Launch IAM policy
+
+Conditions:
+  HasReadWriteAccessArns:
+    !Not [!Equals [!Join [",", !Ref ReadWriteAccessArns], ""]]
+  HasReadOnlyAccessArns:
+    !Not [!Equals [!Join [",", !Ref ReadOnlyAccessArns], ""]]
+  HasAnyAccessArns:
+    !Or [!Condition HasReadWriteAccessArns, !Condition HasReadOnlyAccessArns]
+
+Resources:
+  EncryptionKeyStack:
+    Type: AWS::CloudFormation::Stack
+    Properties:
+      TemplateURL: !Sub ${TemplateRootUrl}/aws-infra/v0.3.1/KMS/kms-key.yaml
+      TimeoutInMinutes: 5
+      Parameters:
+        AliasName: !Sub "alias/${AWS::StackName}-EncryptionKey"
+        AdminPrincipalArns: !Join
+          - ","
+          - - !Sub "arn:aws:iam::${AWS::AccountId}:root"
+            - !Join
+              - ","
+              - !Ref AdminArns
+        UserPrincipalArns: !Join
+          - ","
+          - - !Sub "arn:aws:iam::${AWS::AccountId}:root"
+            - !Join
+              - ","
+              - !Ref AdminArns
+            - !If
+              - HasReadWriteAccessArns
+              - !Join
+                - ","
+                - !Ref ReadWriteAccessArns
+              - !Ref AWS::NoValue
+            - !If
+              - HasReadOnlyAccessArns
+              - !Join
+                - ","
+                - !Ref ReadOnlyAccessArns
+              - !Ref AWS::NoValue
+
+  TowerBucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      BucketName: !Sub "${AWS::StackName}-tower-bucket"
```

Comment:
I'm setting the bucket name because users can no longer use `aws s3 ls` to find the name (since they no longer have `s3:ListAllMyBuckets`).